core/src/software/aws/toolkits/core/credentials/ToolkitCredentialsProvider.kt

```diff
@@ -15,31 +21,25 @@ interface CredentialIdentifier {
      * An optional shortened version of the name to display in the UI where space is at a premium
      */
     val shortName: String get() = displayName
-}
-
-abstract class ToolkitCredentialsIdentifier : CredentialIdentifier {
-    /**
-     * The ID must be unique across all [ToolkitCredentialsIdentifier].
-     * It is recommended to concatenate the factory ID into this field to help enforce this requirement.
-     */
-    abstract val id: String
 
     /**
      * The ID of the corresponding [CredentialProviderFactory] so that the credential manager knows which factory to invoke in order
      * to resolve this into a [ToolkitCredentialsProvider]
      */
-    abstract val factoryId: String
+    val factoryId: String
 
     /**
      * Some ID types (e.g. Profile) have a concept of a default region, this is optional.
      */
-    open val defaultRegionId: String? get() = null
+    val defaultRegionId: String? get() = null
+}
 
+abstract class CredentialIdentifierBase : CredentialIdentifier {
     override fun equals(other: Any?): Boolean {
```

Comment:
we may want to make this explicitly `final` so that it cannot be overridden in an implementer - just did a quick test even though typically `fun` is `final` when we're overriding a base-class method you have to be explicit to stop extenders being able to trump the implementation.

*or* maybe we do want future implementers to be able to have their own equals - I suspect not tho.

Reply:
Good call out, will fix